cdap-master/src/main/java/io/cdap/cdap/master/upgrade/UpgradeJobMain.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.cdap.master.upgrade;
+
+import com.google.common.base.Throwables;
+import io.cdap.cdap.api.retry.RetryableException;
+import io.cdap.cdap.client.ApplicationClient;
+import io.cdap.cdap.client.NamespaceClient;
+import io.cdap.cdap.client.ProgramClient;
+import io.cdap.cdap.client.ScheduleClient;
+import io.cdap.cdap.client.config.ClientConfig;
+import io.cdap.cdap.client.config.ConnectionConfig;
+import io.cdap.cdap.common.NotFoundException;
+import io.cdap.cdap.common.service.Retries;
+import io.cdap.cdap.common.service.RetryStrategies;
+import io.cdap.cdap.common.service.RetryStrategy;
+import io.cdap.cdap.proto.ApplicationRecord;
+import io.cdap.cdap.proto.NamespaceMeta;
+import io.cdap.cdap.proto.ProgramType;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.id.ScheduleId;
+import io.cdap.cdap.proto.id.WorkflowId;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Main class which runs before operator updates CDAP image.
+ * Stops all pipelines and disables all schedules.
+ */
+public class UpgradeJobMain {
+
+  private static final int DEFAULT_READ_TIMEOUT_MILLIS = 90 * 1000;
+  private static final int PORT = 11015;
+  private static final String SCHEDULED = "SCHEDULED";
+
+  public static void main(String[] args) {
+    if (args.length != 1) {
+      throw new RuntimeException(
+        String.format("Invalid number of arguments to UpgradeJobMain. Needed 1, found %d", args.length));
+    }
+    ConnectionConfig connectionConfig = ConnectionConfig.builder()
+      .setHostname("cdap-" + args[0] + "-router")
```

Comment:
this looks very specific to a certain setup, think it's better if the entire hostname is passed in as the arg.